notion/renderer.go

```diff
@@ -0,0 +1,501 @@
+package notion
+
+import (
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/gomarkdown/markdown/ast"
+)
+
+// RenderNodeFunc allows reusing most of Renderer logic and replacing
+// rendering of some nodes. If it returns false, Renderer.RenderNode
+// will execute its logic. If it returns true, Renderer.RenderNode will
+// skip rendering this node and will return WalkStatus
+type RenderNodeFunc func(w io.Writer, node ast.Node) (ast.WalkStatus, bool)
+
+// RendererOptions is a collection of supplementary parameters tweaking
+// the behavior of various parts of HTML renderer.
+type RendererOptions struct {
+	// if set, called at the start of RenderNode(). Allows replacing
+	// rendering of some nodes
+	RenderNodeHook RenderNodeFunc
+}
+
+// Renderer implements Renderer interface for HTML output.
+//
+// Do not create this directly, instead use the NewRenderer function.
+type Renderer struct {
+	opts            RendererOptions
+	TableHolster    TableBlockBlock
+	TableRowHolster TableRowBlockBlock
+	BlocksFinal     []interface{}
```

Comment:
This needs to be uploaded to notion in sets of sub 100 children chunks (due to api limits). this is TODO